docu/SICKAppSpaceCodingStarterKit_Documentation.md

```diff
@@ -667,6 +713,451 @@ Script.serveFunction("CSK_ModuleName.setSelectedInstance", setSelectedInstance)
+```
+
+#### Data Processor
+
+If the feature should receive, process and forward data, this is a kind of combination of the 2 mentioned methods.
+
+Again, create a LUA file like 'ModuleName_ProcessSomething' within the 'FlowConfig' folder and require this within the 'Mainfolder/Subfolder/[ModuleName]_FlowConfig.lua' as well.
+
+This file should include code like this:
+
+```
+-- Block namespace
+local BLOCK_NAMESPACE = 'ModuleName_FC.Process'
+local nameOfModule = 'CSK_ModuleName'
+
+--*************************************************************
+--*************************************************************
+
+-- Required to keep track of already allocated resource
+local instanceTable = {}
+
+local function process(handle, dataSource)
+
+  local instance = Container.get(handle, 'Instance')
+
+  -- Check if amount of instances is valid
+  -- if not: add multiple additional instances
+  while true do
+    local amount = CSK_ModuleName.getInstancesAmount()
+    if amount < instance then
+      CSK_ModuleName.addInstance()
+    else
+      CSK_ModuleName.setInstance(instance)
+      CSK_ModuleName.setRegisterEvent(dataSource)
+      break
+    end
+  end
+
+  -- Here it will check for selected mode. This is only optional
+  local mode = Container.get(handle, 'Mode')
+  if mode == 'TOTAL_SUBRESULTS' then
+    return 'CSK_ModuleName.OnNewStringResult' .. tostring(instance) -- This will return the name of the event, providing relevant data for further processing
+  else
+    return 'CSK_ModuleName.OnNewResult' .. tostring(instance)
+  end
+end
+Script.serveFunction(BLOCK_NAMESPACE .. '.process', process)
+
+--*************************************************************
+--*************************************************************
+
+local function create(instance, mode) -- 'mode' is an extra parameter used in this example but not necessary if not used...
+
+  local fullInstanceName = tostring(instance) .. tostring(mode)
+
+  -- Check if same instance is already configured
+  if instanceTable[fullInstanceName] ~= nil then
+    _G.logger:warning(nameOfModule .. "Instance already in use, please choose another one")
+    return nil
+  else
+    -- Otherwise create handle and store the restriced resource
+    local handle = Container.create()
+    instanceTable[fullInstanceName] = fullInstanceName
+    Container.add(handle, 'Instance', instance)
+    Container.add(handle, 'Mode', mode) -- optional
+    Container.add(handle, "CB_Function", "")
+    return handle
+  end
+end
+Script.serveFunction(BLOCK_NAMESPACE .. '.create', create)
+
+--- Function to reset instances if FlowConfig was cleared
+local function handleOnClearOldFlow()
+  Script.releaseObject(instanceTable)
+  instanceTable = {}
+end
+Script.register('CSK_FlowConfig.OnClearOldFlow', handleOnClearOldFlow)
+
+```
+
+As you can see, now it will register to an event to receive data and will return the name of event itself will provide data, so that other modules can react on that.
```

Comment:
```suggestion
As you can see, now it will register to an event to receive data and will return the name of an event of itself when used to provide data, so that other modules can react on that.
```

Reply:
Done